samples/shopping-cart-addview/src/main/java/shoppingcart/application/CartCloser.java

```diff
@@ -0,0 +1,47 @@
+package shoppingcart.application;
+
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.annotations.Consume;
+import akka.javasdk.client.ComponentClient;
+import akka.javasdk.consumer.Consumer;
+import shoppingcart.application.UserEntity.CloseCartCommand;
+import shoppingcart.domain.ShoppingCartEvent;
+
+// tag::consumer[]
+@ComponentId("cart-closer-consumer")
+@Consume.FromEventSourcedEntity(ShoppingCartEntity.class)
+public class CartCloser extends Consumer {
+
+  private Logger logger = LoggerFactory.getLogger(CartCloser.class);
+  protected final ComponentClient componentClient;
+
+  public CartCloser(ComponentClient componentClient) {
+    this.componentClient = componentClient;
+  }
+
+  // After a user's cart has been checked out, tell the user entity to generate
+  // a new UUID for the current/open cart
+  public Effect onCheckedOut(ShoppingCartEvent.CheckedOut event) {
+    logger.debug("Closing cart for user {} due to checkout", event.userId());
+
+    componentClient.forEventSourcedEntity(event.userId())
+        .method(UserEntity::closeCart)
+        .invokeAsync(new CloseCartCommand(event.cartId()));
+
+    return effects().done();
+  }
+
+  public Effect onItemRemoved(ShoppingCartEvent.ItemRemoved removed) {
+    return effects().ignore();
```

Comment:
you can remove these handlers with: `@Consume.FromEventSourcedEntity(value = ShoppingCartEntity.class, ignoreUnknown = true)`

Reply:
This is so helpful! Wish I'd known about it before